### DISCUSSION TITLE
add machines on network to a docker cluster

### DISCUSSION THREAD

**Discussion by 5at** [siderolabs/talos#8819] [Q&A]
use case: home lab - not production :) and a little out of my depth on some of this so excuse the "newbie"

I have a hefty machine already installed with ubuntu server and would like to use it as the talos cluster control plane.
Would prefereably like to use a talos in a docker container and not use a QEMU machine (that way i dont have to predict how much memory i would need up front, if i have to i will :) )

So i was wondering if there was a way to create a docker version of a cluster and then have other machines on the network join it?

after `talosctl create cluster` i see that the K8s port is bound to '127.0.0.1' - i would probably need to bind it to the main network (10.1.1.0/24) but alas I could not do it (i did use the "-p" switch but that messed up the talosctl create cluster algo)

I also tried [running-talos-in-docker-manually](https://www.talos.dev/v1.7/talos-guides/install/local-platforms/docker/#running-talos-in-docker-manually) - but it kinda left me blank... i think i need a proper config file and setup route tables ?? if this is the way then i have no problem going on an education trip with talos config files.

Any tips appreciated.

PS
Even though i can QEMU, im thinking this is prob not a bad use-case for someone who might want to play / start with talos installation on their home lab and already has a machine without QEMU.. (maybe)

**Comment by smira** [comment]
As I understand, you're running Docker on Linux, in this case there should be no problem whatsoever, you can in fact ignore the port mappings as on your host all Docker IPs (Talos container IPs) are routeable anyways, so you can even use `https://10.5.0.2:6443` as the controlplane endpoint (in default setup, 10.5.0.2 is the first container IP after `talosctl cluster create`).

The problem is getting access from your other machines on your LAN to your docker containers, as by default your host won't route traffic from LAN to your containers. But, you can enable KubeSpan feature on all machines (docker and other VMs), in that way only one-way connection will be enough, and your containers should be able to talk to your LAN hosts, and KubeSpan will offer two-way connectivity on top of that.

So you probably would want to pull back the machine configuration from your docker containers after `talosctl cluster create`, and you can adjust it to match your other machines and apply it there, and they should join the cluster.
